src/shims/alloc.rs

```diff
@@ -109,7 +109,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         // Align must be power of 2, and also at least ptr-sized (POSIX rules).
         // But failure to adhere to this is not UB, it's an error condition.
         if !align.is_power_of_two() || align < this.pointer_size().bytes() {
-            interp_ok(this.eval_libc("EINVAL"))
+            interp_ok(LibcError("EINVAL"))
```

Comment:
Passing a `LibcError("Efoo")` to `interp_ok` results in 
```expected Scalar<Provenance>, found IoError```

Reply:
Yeah leave this unchanged, it can't use the helper. It doesn't set errno anyway.

Reply:
I can revert this, but I'm a little confused. As far as I can see `this.eval_libc` doesn't set the errno, and `LibcError` is just an enum constant. This change would just be for API consistency, right? Just asking in case this is a change you'd want to make eventually.

Reply:
`LibcError` can only be used with `set_last_error`, so only places that set the errno can be converted.